This data set is about identifying if the water quality is good/bad

It is a Labelled dataset with Categorical output, Hence Decision Tree Classification Algorithm will be implemented

Start by importing all necessary modules

In [47]:
import pandas as pd
import numpy as np
from sklearn.metrics import accuracy_score
from sklearn import metrics

Import dataset

In [48]:
data = pd.read_csv('C:\DATAS\milknew.csv')
df=pd.DataFrame(data)
print(df.to_string())

       pH  Temprature  Taste  Odor  Fat   Turbidity  Colour   Grade
0     6.6          35      1     0     1          0     254    high
1     6.6          36      0     1     0          1     253    high
2     8.5          70      1     1     1          1     246     low
3     9.5          34      1     1     0          1     255     low
4     6.6          37      0     0     0          0     255  medium
5     6.6          37      1     1     1          1     255    high
6     5.5          45      1     0     1          1     250     low
7     4.5          60      0     1     1          1     250     low
8     8.1          66      1     0     1          1     255     low
9     6.7          45      1     1     0          0     247  medium
10    6.7          45      1     1     1          0     245  medium
11    5.6          50      0     1     1          1     255     low
12    8.6          55      0     1     1          1     255     low
13    7.4          90      1     0     1        

In [49]:
df['Grade'] = df['Grade'].map({'high': 1, 'low': 0,'medium': 2})


Select features for x and y

In [50]:
x =df.iloc[:, :-1].values # independent variable
y =df.iloc[:, -1].values # dependant variable

Splitting the dataset into the Training set and Test set

In [51]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test= train_test_split(x, y, test_size= 0.25, random_state=0)


Feature Scaling

In [52]:
from sklearn.preprocessing import StandardScaler
st_x= StandardScaler()
x_train= st_x.fit_transform(x_train)
x_test= st_x.transform(x_test)

Fitting Decision Tree classifier to the training set

In [53]:
from sklearn.tree import DecisionTreeClassifier
classifier= DecisionTreeClassifier(criterion='gini', random_state=0)
classifier.fit(x_train, y_train)

DecisionTreeClassifier(random_state=0)

Predicting the Test set results

In [54]:
y_pred= classifier.predict(x_test)
df2=pd.DataFrame({"Actual Y_Test":y_test,"Prediction Data":y_pred})
print("Prediction Result")
print(df2.to_string())

Prediction Result
     Actual Y_Test  Prediction Data
0                1                1
1                0                0
2                2                2
3                0                0
4                1                1
5                2                2
6                0                0
7                1                1
8                1                1
9                2                2
10               0                0
11               1                1
12               2                2
13               2                2
14               1                1
15               0                2
16               1                1
17               1                1
18               1                1
19               0                0
20               1                1
21               0                0
22               2                2
23               1                1
24               1                1
25               0                0
26        

Evaluating Algorithm

In [55]:
print('Mean Absolute Error:', metrics.mean_absolute_error(y_test,y_pred))
print('Mean Squared Error:', metrics.mean_squared_error(y_test,y_pred))
print('Root Mean Squared Error:', np.sqrt(metrics.mean_squared_error(y_test, y_pred)))


Mean Absolute Error: 0.011320754716981131
Mean Squared Error: 0.018867924528301886
Root Mean Squared Error: 0.13736056394868904


Evaluating Predictions

In [56]:
accuracy = accuracy_score(y_test, y_pred)
print('Accuracy: %.2f' % (accuracy*100))

Accuracy: 99.25


The Observed accuracy for this model is 99.25% Hence this is considered as a good model